### PR TITLE
Pa11y

### PR DESCRIPTION
**Pull request by ashwini-mv**
Pa11y-accessibility-tests

### JIRA link ###
https://tools.hmcts.net/jira/browse/SIDM-6723

### Change description ###
Pa11y-accessibility-tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: jest.a11y.config.js [hunk 1]
```diff
@@ -10,4 +10,13 @@ module.exports = {
     '^.+\\.ts?$': 'ts-jest',
   },
   moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
+"reporters": [
```
Comment (KianKwa) [inline jest.a11y.config.js:13]: indentation issue

File: src/test/tsconfig.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
     "esModuleInterop": true,
     "module": "commonjs",
     "strictNullChecks": true,
-    "types": ["jest", "codeceptjs"]
+    "types": ["jest", "codeceptjs","node"],
```
Comment (KianKwa) [inline src/test/tsconfig.json:11]: cery minor issue...missing space between "codeceptjs" and "node" after the comma

File: src/test/a11y/a11y.ts [hunk 3]
```diff
@@ -26,21 +27,13 @@ beforeAll((done /* call it or remove it*/) => {
 });
 
 function ensurePageCallWillSucceed(url: string): Promise<void> {
```
Comment (KianKwa) [inline src/test/a11y/a11y.ts:29]: This function only has one line and it is just calling axios to go to the URL. The function name is "ensurePageCallWillSucceed" and does not reflect what it is doing. I think it is clearer to remove this function and call axios.get(url) directly from the function testAccessibilityWithActions.

File: src/test/a11y/a11y.ts [hunk 4]
```diff
@@ -53,23 +46,27 @@ function expectNoErrors(messages: PallyIssue[]): void {
...
     });
   });
+
+}
+
+function testAccessibility(url: string): void {
+  testAccessibilityWithActions(url, []);
 }
 
 describe('Accessibility', () => {
```
Comment (KianKwa) [inline src/test/a11y/a11y.ts:68]: I know this follows the same style as the FaCT project but I am not keen on the way it is doing the tests within a function. I prefer to separate the tests and the test logic. I think you can use parameterized tests with Jest like below:

const parameters = [
    { url: '/.' },
     // Other URLs go here
  ];

parameters.forEach((parameter) => {
    it('Page \'' + parameter.url + '\' should ave no accessibility errors', done => {
        testAccessibilityWithActions(parameter.url, []);
    });
}

File: src/test/tsconfig.json (hunk 2 above)
Comment (ashwini-mv) [reply src/test/tsconfig.json]: Fixed

File: src/test/a11y/a11y.ts (hunk 4 above)
Comment (ashwini-mv) [reply src/test/a11y/a11y.ts:68]: We discussed about it, decided to keep it as it is for now

**Review by KianKwa — APPROVED** [review]

**Pull request merged**